### ISSUE TITLE
Issue with models.predict() with trained hydrangea CNN

### ISSUE THREAD

**Issue by maulberto3** [keras-team/keras#7959]
Hi friends.
I just trained a hydrangea flower vs. nature images classification app through a CNN: Train and Valid accuracy are ~93%.
**Evaluate generator** gives 95%.
**Predict generator** gives the following individual and _reasonable_ predictions (for 10 hydrangea images and 10 nature images, in separate folders as required by the method), only one incorrect classification assuming the 0.5 threshold:

```
[   0.2020] | [ True]
[   0.0083] | [ True]
[   0.0004] | [ True]
[   0.0125] | [ True]
[   0.0046] | [ True]
[   0.9373] | [*False*]
[   0.0029] | [ True]
[   0.0412] | [ True]
[   0.1569] | [ True]
[   0.1617] | [ True]]

[   0.9631] | [ True]
[   0.9625] | [ True]
[   0.9087] | [ True]
[   0.9652] | [ True]
[   0.9863] | [ True]
[   0.9646] | [ True]
[   0.9638] | [ True]
[   0.9635] | [ True]
[   0.9592] | [ True]
[   0.9679] | [ True]
```

However, when I use the **regular predict**, which does not need the separate image folders as predict generator, then, for 10 random images (out of 100), I get the following individual and _not reasonable_ predictions:

```
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (99).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (29).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (53).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (79).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (83).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (27).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (94).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (48).jpg
1/1 [==============================] - 0s
Prediction is:  0.991576 

data2\part\img (93).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (32).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (60).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (31).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (21).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (12).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (21).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (27).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (23).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (26).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 

data2\part\img (6).jpg
1/1 [==============================] - 0s
Prediction is:  0.0 
```

Since inside I have 50 hydrangea and 50 nature images; then, it does not make sense that **ALL predictions are cero.** 

Any insights on this?

**Comment by harshini-gadige** [comment]
Is this still an issue ?
Are you using tensorflow backend here ?
